cmd/snapshot-controller/main.go

```diff
@@ -87,6 +92,25 @@ func main() {
 	factory := informers.NewSharedInformerFactory(snapClient, *resyncPeriod)
 	coreFactory := coreinformers.NewSharedInformerFactory(kubeClient, *resyncPeriod)
 
+	// Create and register metrics manager
+	metricsManager := metrics.NewMetricsManager()
+	wg := &sync.WaitGroup{}
+	wg.Add(1)
+	if *metricsAddress != "" {
+		srv, err := metricsManager.StartMetricsEndpoint(*metricsPath, *metricsAddress, nil, wg)
```

Comment:
I wasn't sure how the logger was intended to be passed down to the StartMetricsEndpoint? Don't we just use `klog`? Wasn't sure how to pass an instance of klog to the metrics pkg.